Week-3/exercise3.1/memoize.js

```diff
@@ -0,0 +1,27 @@
+function add(a = 0, b = 0) {
+    return a + b;
+}
+
+
+const memoize = (fn) => {
+    const cache = {};
+    return function (a, b) {
+        if (((a + '+' + b) in cache)) {
+            return (cache[a + '+' + b] + ' - no computation');
+        }
+        else {
+            let result = fn(a, b);
+            cache[a + '+' + b] = result;
+            cache[b + '+' + a] = result;
```

Comment:
we can sort the arguments instead of storing it as seperate key

Reply:
function is updated with map